galaxy/main/signals/handlers.py

```diff
@@ -81,9 +77,29 @@ def user_logged_in_handler(request, user, **kwargs):
     }
     provider = models.Provider.objects.get(name__iexact="github")
 
-    models.ProviderNamespace.objects.get_or_create(
```

Comment:
I don't believe this should be changed. There are old namespaces, prior to Galaxy 3, that didn't convert right. In that case the, I think the .get() will fail.

Reply:
Not sure I follow.  Which .get()? 

This line is replaced with either:
``` python
models.ProviderNamespace.objects.get_or_create(name=username,
                                                                                  provider=provider,
                                                                                  defaults=provider_ns_defaults)
```

or 
``` python
models.ProviderNamespace.objects.get_or_create(name=username,
                                                                                  namespace=namespace,
                                                                                  provider=provider,
                                                                                  defaults=provider_ns_defaults)
```

Reply:
I see. You moved the get_or_create below. Nvm.